### PR TITLE
Add ec2 instance options for data and ml nodes

### LINKED ISSUE #6: [FEATURE] Support ML node instance type
### Is your feature request related to a problem?
We see option `mlNodeCount` for ML node count, but we don't see option to specify which EC2 instance type for ML node.

### What solution would you like?
Add one option `mlNodeInstanceType` , so we can specify different instance type to test.

### What alternatives have you considered?
No

### Do you have any additional context?
No

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
This PR adds ec2 isntance options for data and ml nodes. 
This PR also adds the ability to set seed node to data instance type and storage when there is no dedicated master node configured. 
The single-node is fixed to eg.2xlarge due to urgent requirement for it to be used for benchmarking test. Will update this to use the passed data ec2 instance type post P0 release. 
Other nodes except data and ml default to `c5.xlarge`, options for them can be provided in subsequent PRs. 

### Issues Resolved
#6 #27 #28 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by rishabh6788** [comment]
@gaiksaya @peterzhuamazon @prudhvigodithi Need your review.

File: README.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
 | ingestNodeCount (Optional)        | integer | Number of dedicated ingest nodes, default is 0                                                                                                                                                                                                                                                   |
 | mlNodeCount (Optional)            | integer | Number of dedicated machine learning nodes, default is 0                                                                                                                                                                                                                                         |
+| dataInstanceType (Optional)       | string  | Ec2 instance type for data node, defaults to r5.large. See options in `lib/opensearch-config/node-config.ts` for available options. E.g., `-c dataInstanceType=m5.xlarge`                                                                                                                        |
```
Comment (gaiksaya) [inline README.md:54]:
```suggestion
| dataInstanceType (Optional)       | string  | EC2 instance type for data node, defaults to r5.large. See options in `lib/opensearch-config/node-config.ts` for available options. E.g., `-c dataInstanceType=m5.xlarge`                                                                                                                        |
```

File: README.md [hunk 2]
```diff
@@ -51,6 +51,8 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
+| mlInstanceType (Optional)         | string  | Ec2 instance type for ml node, defaults to r5.large. See options in `lib/opensearch-config/node-config.ts` for available options. E.g., `-c mlInstanceType=m5.xlarge`                                                                                                                            |
```
Comment (gaiksaya) [inline README.md:55]:
```suggestion
| mlInstanceType (Optional)         | string  | EC2 instance type for ml node, defaults to r5.large. See options in `lib/opensearch-config/node-config.ts` for available options. E.g., `-c mlInstanceType=m5.xlarge`                                                                                                                            |
```

File: lib/infra/infra-stack.ts [hunk 3]
```diff
@@ -55,6 +55,8 @@ export interface infraProps extends StackProps{
     readonly mlNodeStorage: number,
     readonly jvmSysPropsString?: string,
     readonly additionalConfig?: string,
+    readonly dataEc2InstanceType: InstanceType,
+    readonly mlEc2InstanceType: InstanceType,
```
Comment (gaiksaya) [inline lib/infra/infra-stack.ts:59]: Should we enable instance type for all types of nodes including leader, etc? Keep defaults but allow option to override for others too?

File: README.md (hunk 1 above)
Comment (rishabh6788) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (rishabh6788) [reply README.md]: Done!

File: lib/infra/infra-stack.ts (hunk 3 above)
Comment (rishabh6788) [reply lib/infra/infra-stack.ts:59]: Not required as manager and client nodes not really do any heavy lifting and are only involved in book-keeping and traffic routing tasks. 
We can take it up in future if the need arises. 
@gaiksaya

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**